tensorflow_graphics/projects/mesh_rcnn/ops/vertex_alignment_test.py

```diff
@@ -0,0 +1,162 @@
+                                   image[0, 2, 3, :].numpy()])
+
+    result = vert_align(image, batched_verts, batched_intrinsics)
+
+    self.assertAllClose(expected_result, result[0])
+
+  def test_batch_different_sized_vertices(self):
+    """Tests on batch of different sized vertices."""
+    image = tf.reshape(tf.range(20.), (1, 4, 5, 1))
+    images = tf.repeat(image, repeats=[2], axis=0)
+    verts1 = tf.constant([[-1.5, -1.5, 10],
+                          [-1.5, 0.5, 10],
+                          [-0.5, -1.5, 10],
+                          [-0.5, 0.5, 10]], dtype=tf.float32)
+
+    verts2 = tf.constant([[-1.5, -1.5, 10],
+                          [-1.5, 0.5, 10]], dtype=tf.float32)
+
+    intrinsics = tf.constant([[10, 0, 2.5], [0, 10, 2.5], [0, 0, 1]],
+                             dtype=tf.float32)
+    intrinsics = tf.stack([intrinsics, intrinsics])
+
+    input_mesh = Meshes([verts1, verts2],
+                        [tf.ones((2, 3)), tf.ones((2, 3))])
+
+    expected_result1 = tf.constant([[6.], [8.], [11.], [13.]])
+    expected_result2 = tf.constant([[6.], [8.], [15.], [15.]])
+    result = vert_align(images, input_mesh.get_padded()[0], intrinsics)
+
+    self.assertAllClose(expected_result1, result[0])
+    self.assertAllClose(expected_result2, result[1])
+
+  def test_multiple_batch_dimensions_random_input(self):
+    """Tests on random input with multiple batch dimenaions"""
+    batch_dim_1 = 2
+    batch_dim_2 = 3
+    features = tf.random.normal((batch_dim_1, batch_dim_2, 5, 5, 3))
+    vertices = tf.random.normal((batch_dim_1, batch_dim_2, 10, 3))
+    intrinsics = tf.random.normal((batch_dim_1, batch_dim_2, 3, 3))
+
+    result = vert_align(features, vertices, intrinsics)
+
+    self.assertEqual([batch_dim_1, batch_dim_2, 10, 3], result.shape)
+
+  @parameterized.parameters(
+      ([2, 2, 2, 1], [4, 5, 3], [2, 3, 3],
+       'Not all batch dimensions are identical'),
+      ([4, 2, 2, 1], [2, 5, 3], [2, 3, 3],
+       'Not all batch dimensions are identical'),
+      ([2, 2, 2, 1], [2, 5, 3], [4, 3, 3],
+       'Not all batch dimensions are identical'),
+      ([2, 2, 2, 1], [2, 5, 3], [3],
+       'intrinsics must have a rank greater than 1.'),
+      ([2, 2], [2, 5, 3], [2, 3, 3],
+       'features must have a rank greater than 2.'),
+      ([2, 4, 4, 1], [2, 5, 1], [2, 3, 3],
+       'vertices must have exactly 3 dimensions in axis -1'),
+      ([2, 4, 4, 1], [2, 5, 3], [2, 2, 3],
+       'intrinsics must have exactly 3 dimensions in axis -2'),
+      ([2, 4, 4, 1], [2, 5, 3], [2, 3, 2],
+       'intrinsics must have exactly 3 dimensions in axis -1'),
+  )
+  def test_raising_inputs(self,
+                          feature_shape,
+                          vertex_shape,
+                          intrinsics_shape,
+                          msg):
+    """Tests if correct exceptions are raised when called with wrong
+    input shapes."""
+    features = tf.random.normal(feature_shape)
```

Comment:
FYI self.assert_exception_is_not_raised creates random inputs of correct shapes and calls the method to test for no exceptions.

self.assert_exception_is_raised() does the same and tests for exceptions.